File: 3-й семестр/Lab8/Lab8Sharp/Star.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab8Sharp
{
    public class SellStar
    {
        
        public void sell()
        {
            
            List<string> product = new List<string> { "Water", "Apple", "Milk", "Glue", "Solvent", "gamma-Hydroxybutyric acid", "2,3,7,8-Tetrachlorodibenzodioxin", "1,2-Dichloroethane" };
            List<int> money = new List<int> { 30, 150, 90, 200, 150, 1000, 5000 };
            Random rnd = new Random();
            int chosenProduct = rnd.Next(0, 7);
            Console.Write("The product ");
            Console.Write(product[chosenProduct]);
            Console.Write(" was bought\n");
            earnings += money[chosenProduct];
            Console.Write("The total earnings are ");
            Console.Write(earnings);
            Console.Write("Rub\n\n");
            
        }
        private int earnings = 0;
        public void star()
        {
            SellStar sell = new SellStar();
            for (int j = 0; j < 10; j++)
            {
                ThreadStart threadStart = new ThreadStart(sell.sell);
                Thread thread = new Thread(threadStart);
                thread.Start();
                Thread.Sleep(10);
                
            }
            Console.ReadKey();
        }
    }
}

File: 3-й семестр/Lab8/Lab8/Star.cpp
#include "Star.h"

void SellStar::sell()
{
    
    vector<pair<string, int>> products = { {"Water", 30}, {"Apple", 150}, {"Milk", 90}, {"Glue", 200}, {"Solvent", 150}, {"gamma-Hydroxybutyric acid", 1000} , {"2,3,7,8-Tetrachlorodibenzodioxin", 5000} , {"1,2-Dichloroethane", 110} };
    srand(time(0));
    int chosenProducts = rand() % 7;
    cout << "The product " << products[chosenProducts].first << " was bought" << endl;
    earnings += products[chosenProducts].second;
    cout << "The total earnings are " << earnings << "Rub" << endl << endl;

}
void SellStar::star() {
	
	for (int i = 0; i < 2; i++)
	{
		thread th1([&]() {
			for (int i = 0; i < 5; i++)
			{
				cout << "Cash : ";
				sell();
			}
			});
		
		thread th2([&]() {
			for (int i = 0; i < 5; i++)
			{
				cout << "Cash 2: ";
				sell();
			}
			});
		
		thread th3([&]() {
			for (int i = 0; i < 5; i++)
			{
				cout << "Cash 3: ";
				sell();
			}
			});
		th1.join();
		th2.join();
		th3.join();
	}
	cout << "Time: " << clock();
	system("pause");
}